src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy

```diff
@@ -3598,10 +3598,9 @@ public enum MysteriousTreasures implements LogicCardInfo {
             energyCost R, R, R, R
             attackRequirement {}
             onAttack {
+              discardSelfEnergy R,R
```

Comment:
We now have `discardSelfEnergyAfterDamage` in TcgStatics which will handle discarding energies in an attack in the correct order (announce attack, select energy to discard, perform attack, discard selected energies)

```suggestion
              discardSelfEnergyAfterDamage R,R
```

Reply:
Technically, that doesn't work for this card. The change I made was based on a TODO that's still not entirely complete. https://www.tcgcollector.com/cards/3803/magmortar-lv-x-mysterious-treasures-123-123 The card explicitly requires the energy to be discarded, and then the damage is dealt.